wgsl/index.bs

```diff
@@ -3959,6 +3959,278 @@ TODO: deduplicate these tables
 
 ## Texture built-in functions ## {#texture-builtin-functions}
 
+### Sample ### {#texture-sample-functions}
+
+Sampling from a texture uses the `textureSample` function:
+
+```
+<result> textureSample(<texture>, <sampler>, <coordinates> [, <named-arguments>])
+```
+
+Where:
+
+* `<result>` is the texture's parameterized type when using [sampled](#sampled-texture-type)
+    or [read-only](#texture-ro) texture formats.<br>
+* `<result>` is `f32` when using [depth](#texture-depth) texture types.
+
+* `<texture>` is one of the [sampled](#sampled-texture-type),
+    [read-only](#texture-ro) or [depth](#texture-depth) texture types.
+
+* `<sampler>` is a [sampler type](#sampler-type).
+
+* `<coordinates>` are the texture coordinates used for sampling.
+    The type for the texture coordinates depends on the `<texture>` type and is
+    either a `f32`, `vec2<f32>`, `vec3<f32>` or `vec4<f32>`:
+
+    <table class='data'>
+      <thead>
+        <tr><td>`<texture>` type<td>`<coordinates>` type
+      </thead>
+      <tr><td> `texture_sampled_1d<type>`         <td> `f32`
+      <tr><td> `texture_sampled_1d_array<type>`   <td> `vec2<f32>`
+      <tr><td> `texture_sampled_2d<type>`         <td> `vec2<f32>`
+      <tr><td> `texture_sampled_2d_array<type>`   <td> `vec3<f32>`
+      <tr><td> `texture_sampled_3d<type>`         <td> `vec3<f32>`
+      <tr><td> `texture_sampled_cube<type>`       <td> `vec3<f32>`
+      <tr><td> `texture_sampled_cube_array<type>` <td> `vec4<f32>`
+      <tr><td> `texture_ro_1d<type>`              <td> `f32`
+      <tr><td> `texture_ro_1d_array<type>`        <td> `vec2<f32>`
+      <tr><td> `texture_ro_2d<type>`              <td> `vec2<f32>`
+      <tr><td> `texture_ro_2d_array<type>`        <td> `vec3<f32>`
+      <tr><td> `texture_ro_3d<type>`              <td> `vec3<f32>`
+      <tr><td> `texture_depth_2d`                 <td> `vec2<f32>`
+      <tr><td> `texture_depth_2d_array`           <td> `vec3<f32>`
+      <tr><td> `texture_depth_cube`               <td> `vec3<f32>`
+      <tr><td> `texture_depth_cube_array`         <td> `vec4<f32>`
+    </table>
+
+    Array textures (with `_array` suffix) use the last component of `<coordinates>`
```

Comment:
we need to be more precise here and say that the index in the array is a rounded value of the last component
Perhaps, the group should consider to move this argument out entirely and make it an `u32`?

Reply:
> we need to be more precise here and say that the index in the array is a rounded value of the last component

Done

> Perhaps, the group should consider to move this argument out entirely and make it an u32?

Possibly. HLSL and SPIR-V both use the coordinate packed form. MSL uses a separate `uint` coordinate.
It seems there'll likely be a performance hit for whichever backend we don't match.